app/controllers/discover.js

```diff
@@ -76,15 +77,23 @@ export default Ember.Controller.extend({
         }).then(function(results) {
             var hits = results.aggregations.sources.buckets;
             var providers = [];
+            var whiteList = _this.get('whiteListedProviders');
             hits.map(function(each) {
-                providers.push(each.key);
+                each = each.key;
+                console.log(each);
+                if (whiteList.indexOf(each) !== -1) {
+                    providers.push(each);
+                }
             });
             _this.get('osfProviders').slice().map(function(each) {
```

Comment:
I know this is not part of PR but maybe you can join these two provider lists and do this step only once?